ndbench-dynamodb-plugins/src/main/java/com/netflix/ndbench/plugin/dynamodb/DynamoDBKeyValue.java

```diff
@@ -0,0 +1,233 @@
+				new ProvisionedThroughput(this.config.getReadCapacityUnits(), this.config.getWriteCapacityUnits()));
+
+		logger.debug("Waiting until the table is in ACTIVE state");
+		table.waitForActive();
+		
+		DescribeTableRequest describeTableRequest = new DescribeTableRequest()
+				.withTableName(this.config.getTableName());
+		TableDescription tableDescription = client.describeTable(describeTableRequest).getTable();
+		logger.info("Table Description: " + tableDescription);
+
+		logger.info("DynamoDB Plugin initialized");
+	}
+
+	/**
+	 * 
+	 * @param key
+	 * @return the item
+	 * @throws Exception
+	 */
+	@Override
+	public String readSingle(String key) throws Exception {
+		Item item = null;
+		try {
+			GetItemSpec spec = new GetItemSpec().withPrimaryKey("Id", key).withConsistentRead(config.consistentRead());
+			item = table.getItem(spec);
+			if (item == null) {
+				return null;
+			}
+		} catch (AmazonServiceException ase) {
+			amazonServiceException(ase);
+		} catch (AmazonClientException ace) {
+			amazonClientException(ace);
+		}
+		return item.toString();
+	}
+
+	/**
+	 * 
+	 * @param key
+	 * @return A string representation of the output of a PutItemOutcome operation.
+	 * @throws Exception
+	 */
+	@Override
+	public String writeSingle(String key) throws Exception {
+		PutItemOutcome outcome = null;
+		try {
+			Item item = new Item().withPrimaryKey("Id", key).withString("value", this.dataGenerator.getRandomValue());
+			// Write the item to the table
+			outcome = table.putItem(item);
+			if (outcome == null) {
+				return null;
+			}
+
+		} catch (AmazonServiceException ase) {
+			amazonServiceException(ase);
+		} catch (AmazonClientException ace) {
+			amazonClientException(ace);
+		}
+		return outcome.toString();
+	}
+
+	@Override
+	public List<String> readBulk(List<String> keys) throws Exception {
+		return null;
+	}
+
+	@Override
+	public List<String> writeBulk(List<String> keys) throws Exception {
+		return null;
+	}
+
+	@Override
+	public void shutdown() throws Exception {
+		try {
+			logger.info("Issuing DeleteTable request for " + config.getTableName());
+			table.delete();
+
+			logger.info("Waiting for " + config.getTableName() + " to be deleted...this may take a while...");
+
+			table.waitForDelete();
+		} catch (Exception e) {
+			logger.error("DeleteTable request failed for " + config.getTableName());
+			logger.error(e.getMessage());
+		}
+		table.delete(); // cleanup
+		client.shutdown();
+		logger.info("DynamoDB shutdown");
+	}
+
+	/*
+	 * Not needed for this plugin
+	 * 
+	 * @see com.netflix.ndbench.api.plugin.NdBenchClient#getConnectionInfo()
+	 */
+	@Override
+	public String getConnectionInfo() throws Exception {
+		return null;
+	}
+
+	@Override
+	public String runWorkFlow() throws Exception {
+		return null;
+	}
+
+	private void amazonServiceException(AmazonServiceException ase) {
+
+		logger.error("Caught an AmazonServiceException, which means your request made it "
```

Comment:
DO you want to log a single message or 6 different messages here?

Reply:
This is standard AWS way of reporting errors. In fact, each message presents some information that is useful.

Reply:
Agreed. I was just wondering if it is needed for any reason (like grepping) to have the entire message in a single line rather than split across 6 lines. I am fine either ways.